### PR TITLE
vlan dot1q support

### PR DESCRIPTION
**Pull request by prashanthjbabu**
This commit adds DOT1Q(VLAN) support so that JOY maybe able to parse such packets! This handles Issue -  "802.1Q Virtual LAN #77"

### CODE REVIEW

**Comment by fliphil** [comment]
Could you please supply us with the PCAP file that you used to test? If it's not private you can throw it into the "resources" directory. Otherwise, you could email it directly.

File: src/pkt_proc.c [hunk 1]
```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
...
+           vlan_ether_type = ntohs(*(uint16_t *)(packet + ETHERNET_HDR_LEN + 2));
+           switch(vlan_ether_type) {
+               case ETH_TYPE_IP:
+                   ip = (struct ip_hdr*)(packet + ETHERNET_HDR_LEN + DOT1Q_HDR_LEN);
+                   ip_hdr_len = ip_hdr_length(ip);
+                   break;
+           default :
+               return;
+           }
+       break;
```
Comment (fliphil) [inline src/pkt_proc.c]: please indent this break 1 time to the right.

File: src/pkt_proc.c [hunk 2]
```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
...
-  
-    /* define/compute ip header offset */
-    ip = (struct ip_hdr*)(packet + ETHERNET_HDR_LEN);
-    ip_hdr_len = ip_hdr_length(ip);
+    ether_type = ntohs(*(uint16_t *)(packet + 12));//Offset to get ETH_TYPE
+    switch(ether_type) {
+       case ETH_TYPE_IP:
+           ip = (struct ip_hdr*)(packet + ETHERNET_HDR_LEN);
+           ip_hdr_len = ip_hdr_length(ip);
+       break;
```
Comment (fliphil) [inline src/pkt_proc.c]: please indent this break 1 time to the right.

File: src/pkt_proc.c [hunk 3]
```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
...
+       case ETH_TYPE_DOT1Q:
+           //Offset to get VLAN_TYPE
+           vlan_ether_type = ntohs(*(uint16_t *)(packet + ETHERNET_HDR_LEN + 2));
+           switch(vlan_ether_type) {
+               case ETH_TYPE_IP:
+                   ip = (struct ip_hdr*)(packet + ETHERNET_HDR_LEN + DOT1Q_HDR_LEN);
+                   ip_hdr_len = ip_hdr_length(ip);
+                   break;
+           default :
```
Comment (fliphil) [inline src/pkt_proc.c]: please indent this default block 1 time to the right.

File: src/pkt_proc.c [hunk 4]
```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
...
     fprintf(info, "\n");
     joy_log_info("++++++++++ Packet %lu ++++++++++", flocap_stats_get_num_packets());
     //  packet_count++;
   
     // ethernet = (struct ethernet_hdr*)(packet);
-  
-    /* define/compute ip header offset */
-    ip = (struct ip_hdr*)(packet + ETHERNET_HDR_LEN);
-    ip_hdr_len = ip_hdr_length(ip);
+    ether_type = ntohs(*(uint16_t *)(packet + 12));//Offset to get ETH_TYPE
```
Comment (fliphil) [inline src/pkt_proc.c]: A comment here explaining the purpose of this code block, i.e. "Support of both normal ethernet and 802.1q. Distinguish between the two accepted types."

File: src/pkt_proc.c [hunk 5]
```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
...
+    switch(ether_type) {
+       case ETH_TYPE_IP:
+           ip = (struct ip_hdr*)(packet + ETHERNET_HDR_LEN);
+           ip_hdr_len = ip_hdr_length(ip);
+       break;
+       case ETH_TYPE_DOT1Q:
+           //Offset to get VLAN_TYPE
```
Comment (fliphil) [inline src/pkt_proc.c]: please place joy_log_info("Ethernet type - 802.1q vlan")

File: src/pkt_proc.c [hunk 6]
```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
...
     //  packet_count++;
   
     // ethernet = (struct ethernet_hdr*)(packet);
-  
-    /* define/compute ip header offset */
-    ip = (struct ip_hdr*)(packet + ETHERNET_HDR_LEN);
-    ip_hdr_len = ip_hdr_length(ip);
+    ether_type = ntohs(*(uint16_t *)(packet + 12));//Offset to get ETH_TYPE
+    switch(ether_type) {
+       case ETH_TYPE_IP:
```
Comment (fliphil) [inline src/pkt_proc.c]: place joy_log_info("Ethernet type - normal")

File: src/pkt_proc.c (hunk 4 above)
Comment (prashanthjbabu) [inline src/pkt_proc.c]: Thanks! Comments addressed in new commit !

**Comment by prashanthjbabu** [comment]
Thanks Philip! i've addressed your comments in a new commit!

**Comment by fliphil** [comment]
#83

**Pull request merged**